yt_dlp/extractor/goplay.py

```diff
@@ -0,0 +1,76 @@
+import json
+
+from .common import InfoExtractor
+from ..utils import int_or_none
+from .goplay_auth_aws import AwsIdp
+
+
+class GoPlayIE(InfoExtractor):
+    _VALID_URL = r'https?://(www\.)?goplay\.be/video/(?P<series>[^/]+)/(?P<season>[^/]+)(/(?P<episode>[^/]+)|)'
+
+    _NETRC_MACHINE = 'goplay'
+
+    _TESTS = [{
+        'url': 'https://www.goplay.be/video/de-container-cup/de-container-cup-s3/de-container-cup-s3-aflevering-2#autoplay',
+        'info_dict': {
+            'id': '9c4214b8-e55d-4e4b-a446-f015f6c6f811',
+            'ext': 'mp4',
+            'title': 'S3 - Aflevering 2',
+            'series': 'De Container Cup',
+            'season': 'Season 3',
+            'season_number': 3,
+            'episode': 'Episode 2',
+            'episode_number': 2,
+        },
+        'skip': 'This video is only available for registered users'
+    }]
+
+    def _initialize_pre_login(self):
+        self._id_token = None
+
+    def _perform_login(self, username, password):
+        self.report_login()
+        aws = AwsIdp(pool_id='eu-west-1_dViSsKM5Y', client_id='6s1h851s8uplco5h6mqh1jac8m')
+        self._id_token, _ = aws.authenticate(username=username, password=password)
+
+    def _real_initialize(self):
+        if not self._id_token:
+            raise self.raise_login_required(method='password')
+
+    def _real_extract(self, url):
+        if "#" in url:
+            url = url.split("#")[0]
+
+        webpage = self._download_webpage(url, None)
+
+        video_data = self._html_search_regex(r'<div data-hero="([^"]*)"', webpage, 'video_data')
+        playlists = json.loads(video_data.replace('&quot;', '"'))['data']['playlists']
+        wanted_playlist = [x for x in playlists if x['pageInfo']['url'] in url][0]
+        wanted_episode = [x for x in wanted_playlist['episodes'] if x['pageInfo']['url'] == url][0] or [x for x in wanted_playlist['episodes'] if x['pageInfo']['url'] in url][0]
+
+        title = wanted_episode['episodeTitle']
+        series = wanted_episode['program']['title']
+        season = wanted_episode['seasonNumber']
+        episode = wanted_episode['episodeNumber']
+        video_id = wanted_episode['videoUuid']
+
+        api_url = 'https://api.viervijfzes.be/content/%s' % video_id
+        api_headers = {
+            'authorization': self._id_token,
+        }
+        api = self._download_json(api_url, video_id, headers=api_headers)
+
+        formats = self._extract_m3u8_formats(
+            api['video']['S'], video_id, ext='mp4',
+            entry_protocol='m3u8_native', m3u8_id='HLS', fatal=False
+        )
+        self._sort_formats(formats)
+
+        return {
+            'id': video_id,
+            'title': title,
+            'series': series,
+            'season_number': int_or_none(season),
+            'episode_number': int_or_none(episode),
+            'formats': formats,
+        }
```

Comment:
```py
        video_id = episode['videoUuid']
        api = self._download_json(
            f'https://api.viervijfzes.be/content/{video_id}',
             video_id, headers={'Authorization': self._id_token})

        formats, subs = self._extract_m3u8_formats_and_subtitles(
            api['video']['S'], video_id, ext='mp4', m3u8_id='HLS', fatal=False)
        self._sort_formats(formats)

        return {
            'id': video_id,
            'title': episode.get('episodeTitle'),
            ...
        }
```

Reply:
Thanks a lot! That's way better. I've updated the code to reflect this.